tests/bin/check-heap-changes.py

```
@@ -0,0 +1,119 @@
+    Parse input file and return a map of object class name to count.
+    '''
+    data = {}
+    try:
+        with open(filename, 'r') as f:
+            for line in f:
+                line = line.strip()
+                if not line:
+                    continue
+                parts = line.split()
+                if len(parts) != 2:
+                    print(f'ERROR: File {filename} line {line_num} does not contain exactly 2 columns', file=sys.stderr)
```

Ruff (0.16.0)

[error] Undefined name `line_num`

(F821)